testworld/signature_test.go

```diff
@@ -0,0 +1,88 @@
+// +build testworld
+
+package testworld
+
+import (
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/testingutils/config"
+	"github.com/centrifuge/go-centrifuge/testingutils/documents"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+func TestHost_FakedSignature(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+	bob := doctorFord.getHostTestSuite(t, "Bob")
+	eve := doctorFord.getHostTestSuite(t, "Eve")
+
+	ectxh := testingconfig.CreateAccountContext(t, eve.host.config)
+
+	collaborators := [][]byte{bob.id[:]}
+	coredoc := prepareCoreDocument(t, collaborators, eve.host.docSrv, alice.host.config)
+
+	err := eve.host.p2pClient.GetSignaturesForDocument(ectxh, coredoc)
+	assert.NoError(t, err)
```

Comment:
with new changes to develop, you should be able to obtain the actual error by bob

Reply:
Yeah, with the latest changes, lot of stuff has been broken. I'll update this PR once I've fixed everything and the tests work as expected :)